cuda_core/tests/test_launcher.py

```diff
@@ -59,6 +59,60 @@ def test_launch_config_shmem_size():
     assert config.shmem_size == 0
 
 
+def test_launch_config_cluster_grid_conversion():
```

Comment:
This test is way more complex than it could be. We should not need to mock anything. 
- take the `init_cuda` fixture as test input
- keep test case 1-5 you have strip out everything else
- keep the test cases under a try-except: if `CUDAError` is raised, skip the test via `pytest.skip` (it's either the driver or the GPU is not new enough)